### DISCUSSION TITLE
Nested filtering on screens based on related tables

### DISCUSSION THREAD

**Discussion by bmscmoreira** [Budibase/budibase#12313] [Help]
I am using the internal database an I have 4 tables:
- a FLOORS table with FLOOR_NAME text field and a relationship to a SPACES table;
- a SPACES table with SPACE_NAME text field and a relationship to a ASSETS table;
- an ASSETS table with ASSET_NAME text field.
- an EVENTS table with EVENT_DATE date field and relationships to FLOORS, SPACES and ASSETS table;

The logic is that an ASSET belongs to a one SPACE which, in turn, belongs to one FLOOR.

I'd like to create a frontend screen where the user:
- first selects a FLOOR;
- then selects a SPACE from all available spaces in that floor;
- then selects an ASSET from all available assets in that space.
- the hits a submit button and this information would be stored in the EVENTS table, together with the current date stored in the EVENT_DATE date field.

Still learning and trying with several components (forms, tables, options filters, relationship filters) but no luck so far. All help appreciated!

**Comment by melohagan** [comment]
Hey @bmscmoreira 

Did you see these docs pages at all? They may help to get you started:

- https://docs.budibase.com/docs/filter-by-relationships#tutorial-filter-on-selected-relationship-picker-values
- https://docs.budibase.com/docs/cascading-dropdown-filters
